src/main/java/com/microsoft/azure/vmagent/util/Constants.java

```diff
@@ -240,7 +240,7 @@ public final class Constants {
 
     public static final String DEFAULT_SUBNET_NAME = "jenkinsarm-snet";
 
-    public static final String DEFAULT_RESOURCE_GROUP_PATTERN = "^[a-zA-Z0-9][a-zA-Z\\-_0-9]{0,62}[a-zA-Z0-9]$";
+    public static final String DEFAULT_RESOURCE_GROUP_PATTERN = "^[a-zA-Z0-9][a-zA-Z\\-\\._0-9]{0,62}[a-zA-Z0-9]$";
```

Comment:
this is the minimal change, the other bits of this PR just make it more future proof